dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -70,66 +71,164 @@ class Identity[A] extends Transformer[A, A] {
   }
 }
 
+/**
+ * convert sample to MiniBatch, and you can choose whether padding feature or label
+ * to same length in a batch
+ */
 object SampleToBatch {
   def apply[T: ClassTag]
-  (batchSize: Int)
+  (batchSize : Int,
+   featurePadding : Option[Tensor[T]] = None,
+   labelPadding : Option[T] = None,
+   fixedLength: Option[Int] = None)
```

Comment:
featurePadding  feature padding value (by default None, meaning no feature padding)

labelPadding label padding value (by default None, meaning no label padding)

fixedLength if padding, it specifies the length of feature/label after padding (by default None, meaning the length after padding is set to the max length of feature/label in a mini-batch)